pkg/experiment/symbolizer/symbolizer.go

```diff
@@ -109,12 +115,30 @@ func NewProfileSymbolizer(logger log.Logger, client DebuginfodClient, store Debu
 		client:          client,
 		store:           store,
 		metrics:         metrics,
+		limits:          limits,
 		symbolCache:     symbolCache,
 		lidiaTableCache: lidiaTableCache,
 	}, nil
 }
 
 func (s *ProfileSymbolizer) SymbolizePprof(ctx context.Context, profile *googlev1.Profile) error {
+	tenantIDs, err := tenant.TenantIDs(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get tenant ID: %w", err)
+	}
+	if len(tenantIDs) > 1 {
+		return fmt.Errorf("symbolization does not support multi-tenant requests")
+	}
+	tenantID := tenantIDs[0]
+
+	if !s.limits.SymbolizerEnabled(tenantID) {
+		level.Debug(s.logger).Log(
+			"msg", "symbolization disabled for tenant",
+			"tenant", tenantID,
+		)
+		return nil
+	}
+
```

Comment:
I'm wondering if the check should be performed by the `SymbolizePprof` caller

Reply:
Initially I considered moving this logic to the query frontend level, where `SymbolizePprof` is called. However, this would require either adding symbolizer limits to frontend limits or passing additional limits to the query frontend. I opted to keep it in the symbolizer to maintain encapsulation, though moving the check outside remains a valid alternative if we want to make symbolization decisions more explicit. 
Do you think the code would be more readable?  I can do the change, since I wasn't sure about it neither.

Reply:
I suggest that we perform the check on the caller side. It's more explicit, and it's how such checks are implemented in other places

Reply:
Updated in 821610f57e328cda7f02da29c39c8d9a2682402f